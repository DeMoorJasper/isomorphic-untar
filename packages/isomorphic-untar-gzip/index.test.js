const fs = require("fs-extra");
const path = require("path");

const untar = require("./lib/node").default;

const FIXTURES_DIR = path.join(__dirname, "fixtures");

describe("isomorphic-untar", () => {
  it("Should be able to extract a tar file", async () => {
    let tarFile = await fs.readFile(
      path.join(FIXTURES_DIR, "react-17.0.2.tar")
    );
    let files = await untar(tarFile.buffer);
    expect(files).toMatchSnapshot();
  });

  it("Should be able to extract a tar.gz file", async () => {
    let tarFile = await fs.readFile(
      path.join(FIXTURES_DIR, "react-17.0.2.tgz")
    );
    let files = await untar(tarFile.buffer);
    expect(files).toMatchSnapshot();
  });

  it("Should also work with a UInt8Array", async () => {
    let tarFile = await fs.readFile(
      path.join(FIXTURES_DIR, "react-17.0.2.tgz")
    );
    let files = await untar(new Uint8Array(tarFile.buffer));
    expect(files).toMatchSnapshot();
  });
});
