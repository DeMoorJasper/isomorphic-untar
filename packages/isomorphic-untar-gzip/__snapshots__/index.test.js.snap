// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isomorphic-untar Should be able to extract a tar file 1`] = `
Array [
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/LICENSE",
    "size": 1086,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/index.js",
    "size": 190,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/jsx-dev-runtime.js",
    "size": 222,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/jsx-runtime.js",
    "size": 214,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react-jsx-dev-runtime.development.js",
    "size": 37753,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react-jsx-dev-runtime.production.min.js",
    "size": 456,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react-jsx-dev-runtime.profiling.min.js",
    "size": 455,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react-jsx-runtime.development.js",
    "size": 38352,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react-jsx-runtime.production.min.js",
    "size": 962,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react-jsx-runtime.profiling.min.js",
    "size": 961,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react.development.js",
    "size": 72141,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/umd/react.development.js",
    "size": 105096,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react.production.min.js",
    "size": 6450,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/umd/react.production.min.js",
    "size": 11440,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/umd/react.profiling.min.js",
    "size": 13668,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/build-info.json",
    "size": 167,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/package.json",
    "size": 777,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/README.md",
    "size": 737,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
]
`;

exports[`isomorphic-untar Should be able to extract a tar.gz file 1`] = `
Array [
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/LICENSE",
    "size": 1086,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/index.js",
    "size": 190,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/jsx-dev-runtime.js",
    "size": 222,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/jsx-runtime.js",
    "size": 214,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react-jsx-dev-runtime.development.js",
    "size": 37753,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react-jsx-dev-runtime.production.min.js",
    "size": 456,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react-jsx-dev-runtime.profiling.min.js",
    "size": 455,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react-jsx-runtime.development.js",
    "size": 38352,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react-jsx-runtime.production.min.js",
    "size": 962,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react-jsx-runtime.profiling.min.js",
    "size": 961,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react.development.js",
    "size": 72141,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/umd/react.development.js",
    "size": 105096,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/cjs/react.production.min.js",
    "size": 6450,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/umd/react.production.min.js",
    "size": 11440,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/umd/react.profiling.min.js",
    "size": 13668,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/build-info.json",
    "size": 167,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/package.json",
    "size": 777,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
  Object {
    "buffer": ArrayBuffer [],
    "devmajor": 0,
    "devminor": 0,
    "gid": 0,
    "gname": "",
    "mode": 420,
    "mtime": 499162500000,
    "name": "package/README.md",
    "size": 737,
    "type": "file",
    "uid": 0,
    "uname": "",
  },
]
`;
